### ISSUE TITLE
OData-Version response header false negative @REQ-WA103-RC11

### ISSUE THREAD

**Issue by dgmyrek** [RESOStandards/web-api-commander#15]
This test is failing due to an apparent missing OData-Version header, however, I can confirm that our server is including "odata-version: 4.0" in the response headers for this exact request (see excerpt from logs at bottom). I'm guessing the check for that response header just needs to be made case insensitive.

@REQ-WA103-RC11 @core @2.5.2 @core-endorsement
Scenario: REQ-WA103-RC11 - 501 Not Implemented Request                                       # src/main/java/org/reso/certification/features/web-api/web-api-server-1.0.2.feature:297
Using RESOScript: generic.gold.resoscript
  Given a RESOScript file was provided                                                       # org.reso.certification.stepdefs.WebAPIServer_1_0_2.<init>(WebAPIServer_1_0_2.java:73)
RESOScript loaded successfully!
  And Client Settings and Parameters were read from the file                                 # org.reso.certification.stepdefs.WebAPIServer_1_0_2.<init>(WebAPIServer_1_0_2.java:80)
Bearer token loaded... first 4 characters: <redacted>
Service root is: https://api.flexmls.com/Reso/OData
  And a test container was successfully created from the given RESOScript                    # org.reso.certification.stepdefs.WebAPIServer_1_0_2.<init>(WebAPIServer_1_0_2.java:88)
Authentication Type: authorization_code
  And the test container uses an authorization_code or client_credentials for authentication # org.reso.certification.stepdefs.WebAPIServer_1_0_2.<init>(WebAPIServer_1_0_2.java:95)
Request ID: REQ-WA103-RC11
Request URI: https://api.flexmls.com/Reso/OData/Property?$search=red%20OR%20blue
  When a GET request is made to the resolved Url in "REQ-WA103-RC11"                         # org.reso.certification.stepdefs.WebAPIServer_1_0_2.<init>(WebAPIServer_1_0_2.java:311)
Asserted Response Code: 501, Server Response Code: 501
  Then the server responds with a status code of 501                                         # org.reso.certification.stepdefs.WebAPIServer_1_0_2.<init>(WebAPIServer_1_0_2.java:316)
  And the server has an OData-Version header value of "4.0" or "4.01"                        # org.reso.certification.stepdefs.WebAPIServer_1_0_2.<init>(WebAPIServer_1_0_2.java:1019)
      java.lang.AssertionError: ERROR: must specify an 'OData-Version' in the response header!
See: http://docs.oasis-open.org/odata/odata/v4.0/errata03/os/complete/part1-protocol/odata-v4.0-errata03-os-part1-protocol-complete.html#_Toc453752225
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.junit.Assert.assertNotNull(Assert.java:713)
        at org.reso.certification.stepdefs.WebAPIServer_1_0_2.lambda$new$65(WebAPIServer_1_0_2.java:1023)
        at ?.the server has an OData-Version header value of "4.0" or "4.01"(file:web-api-commander-master/src/main/java/org/reso/certification/features/web-api/web-api-server-1.0.2.feature:300)

[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << HTTP/1.1 501 Not Implemented
[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << Server: nginx
[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << Date: Mon, 06 Apr 2020 18:35:42 GMT
[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << Content-Type: application/json;odata.metadata=minimal
[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << Connection: keep-alive
[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << cache-control: no-cache

[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << x-xss-protection: 1; mode=block
[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << x-runtime: 0.034894
[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << x-content-type-options: nosniff
[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << odata-version: 4.0
[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << x-powered-by: Phusion Passenger 5.0.6
[2020-04-06 13:35:40.572] [DEBUG] [headers] - http-outgoing-30 << status: 501 Not Implemented

**Comment by darnjo** [comment]
Interesting. I thought the case-insensitivity was already being enforced by the test. I'll double-check and circle back on the issue today so we can get it wrapped up.

**Comment by darnjo** [comment]
This turned out to be a bug in the Olingo library. It's not returning headers when this particular kind of error is thrown, so trying to access them returns null and the test fails. [I found an issue where headers were added to other exceptions in OData](https://issues.apache.org/jira/browse/OLINGO-1380) and asked whether they could be added in this case as well. I can make the changes myself and open a PR with Olingo if need be.
